### PR TITLE
Added new Quantile method.

### PR DESCRIPTION
**Pull request by marc-opengamma**
Created a new quantile method with exponentially weighted probabilities.
The class implement new methods with output type 'QuantileResult'. The result contains the information about the quantile itself but also how it was computed, with which scenarios and which weight. This type of result could be extended to other quantile methods if appropriate for other cases.

### CODE REVIEW

File: modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/ExponentiallyWeightedInterpolationQuantileMethod.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+  @Override
+  protected double expectedShortfall(double level, DoubleArray sortedSample) {
+    throw new UnsupportedOperationException("Expected Shortfall only from unsorted sample due to weights.");
+  }
+  
+  /**
+   * Returns the weights for a given sample size.
+   * @param size  the sample size
+   * @return teh weights
```
Comment (yukiiwashita) [inline modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/ExponentiallyWeightedInterpolationQuantileMethod.java]: typo

File: modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/ExponentiallyWeightedInterpolationQuantileMethod.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  
+  /** The exponential weight. */
+  private final double lambda;
+
+  /**
+   * Constructor. 
+   * @param lambda  the exponential weight
+   */
+  public ExponentiallyWeightedInterpolationQuantileMethod(double lambda) {
+    this.lambda = lambda;
```
Comment (yukiiwashita) [inline modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/ExponentiallyWeightedInterpolationQuantileMethod.java]: Any constraint on the range of lambda? The formula calculating weights breaks down in some cases.

File: modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/ExponentiallyWeightedInterpolationQuantileMethod.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    return q.getValue();
+  }
+  
+  @Override
+  public double expectedShortfallFromUnsorted(double level, DoubleArray sample) {
+    QuantileResult q = quantileDetails(level, sample, true, true);
+    return q.getValue();
+  }
+  
+  public QuantileResult quantileDetailsFromUnsorted(double level, DoubleArray sample) {
```
Comment (yukiiwashita) [inline modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/ExponentiallyWeightedInterpolationQuantileMethod.java]: Add javadoc for the public method.

File: modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/ExponentiallyWeightedInterpolationQuantileMethod.java [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    return quantileDetails(level, sample, true, false);
+  }
+  
+  public QuantileResult expectedShortfallDetailsFromUnsorted(double level, DoubleArray sample) {
```
Comment (yukiiwashita) [inline modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/ExponentiallyWeightedInterpolationQuantileMethod.java]: Add javadoc for the public method.

File: modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/QuantileResult.java [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+import org.joda.beans.Property;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+/**
+ * Object describing the result from a {@link QuantileCalculationMethod}.
+ */
+@BeanDefinition
```
Comment (jodastephen) [inline modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/QuantileResult.java]: Use `builderScope="private"`

File: modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/QuantileResult.java [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+public final class QuantileResult
+  implements ImmutableBean, Serializable {
+  
+  /** The quantile value. */
+  @PropertyDefinition
+  private final double value;
+  /** The indices of sample data used in the calculation. For discrete methods, the length is 1 and for interpolated
+   * methods, the length is usually 2. */
+  @PropertyDefinition
+  private final int[] indices;
```
Comment (jodastephen) [inline modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/QuantileResult.java]: Is it ever greater than 2? If not, maybe it is one `int` and one optional `int`.

Comment (marc-opengamma) [inline modules/math/src/main/java/com/opengamma/strata/math/impl/statistics/descriptive/QuantileResult.java]: I most of the implementation we have it is 1 or 2. But we could imagine interpolator that are non-local or quantile measurement by Extreme Value Theory were a large range of indices would be used. In the new quantile method implementation, the "expectedShortfallFromUnsorted" uses a range of indices.

**Pull request merged**